mathjax3-ts/output/chtml/Wrappers/mtable.ts

```diff
@@ -0,0 +1,387 @@
+            cSpace.map(x => parseFloat(x)).reduce((a, b) => a + b) + 2 * parseFloat(fSpace[1]);
+        return this.bbox;
+    }
+
+    /******************************************************************/
+
+    /*
+     * Pad any short rows with extra cells
+     */
+    protected padRows() {
+        for (const row of Array.from(this.chtml.childNodes)) {
+            while (row.childNodes.length < this.numCols) {
+                row.appendChild(this.html('mjx-mtd'));
+            }
+        }
+    }
+
+    /*
+     * Set the cell aligmentsbased on the table, row, or cell columnalign attributes
+     */
+    protected handleColumnAlign() {
+        const colAlign = this.getColumnAttributes('columnalign') || [];
+        for (const row of this.childNodes) {
+            const aligns = this.getColumnAttributes('columnalign', row) || colAlign;
+            let i = 0;
+            for (const cell of row.childNodes) {
+                let align = (cell.node.attributes.get('columnalign') as string) || aligns[i++];
+                if (align !== 'center') {
+                    cell.chtml.style.textAlign = align;
+                }
+            }
+        }
+    }
+
+    /*
+     * Set the inter-column spacing for all columns
+     *  (Use frame spacing on the outsides, if needed, and use half the column spacing on each
+     *   neighboring column, so that if column lines are needed, they fall in the middle
+     *   of the column space.)
+     *
+     * @param{boolean} frame  Whether to include frame spacing on the left and right or not
+     * @param{string} fspace  The frame spacing to use, if any
+     */
+    protected handleColumnSpacing(frame: boolean, fspace: string) {
+        //
+        //  Get the column spacing values, and add the frame spacing values at the left and right
+        //
+        const spacing = this.convertLengths(this.getColumnAttributes('columnspacing'), 2);
+        if (!spacing) return;
+        spacing.unshift(fspace);
+        spacing[this.numCols] = fspace;
+        //
+        //  For each row...
+        //
+        for (const row of this.childNodes) {
+            let i = 0;
+            //
+            //  For each cell in the row...
+            //
+            for (const cell of row.childNodes) {
+                //
+                //  Get the left and right-hand spacing
+                //
+                const lspace = spacing[i++];
+                const rspace = spacing[i];
+                //
+                //  Set the style for the spacing, if it is needed, and isn't the
+                //  default already set in the mtd styles
+                //
+                const style = (cell ? cell.chtml : row.chtml.childNodes[i] as HTMLElement).style;
+                if ((i > 1 || frame) && lspace !== '.5em') {
+                    style.paddingLeft = lspace;
+                }
+                if ((i < this.numCols || frame) && rspace !== '.5em') {
```

Comment:
Sanity check: If in a frame than a right space element will exist?

Reply:
Yes, it is added to the spacing array at line 191 above.  I've also made changes to guarantee it is defined.